app/sjson/memcompat.h

```diff
@@ -4,8 +4,4 @@
 #include "c_types.h"
 #include "mem.h"
 
```

Comment:
Excellent work! When I was testing build with newer xtensa toolchain those were a real pain. Just few things:
Maybe it would be a good idea to just remove this header file and it's occurrence in [Makefile](https://github.com/nodemcu/nodemcu-firmware/blob/2f70d4efcc28944138e144c7cc041b85812cd436/app/sjson/Makefile#L26)?

Reply:
It should certainly be possible to adjust the sjson module to not depend on the SDK specific things; at the time I went with minimum possible change just to get through the bulk of it.